web_widget_x2many_2d_matrix/static/src/xml/web_widget_x2many_2d_matrix.xml

```diff
@@ -0,0 +1,36 @@
+<templates>
+    <t t-name="FieldX2Many2dMatrix">
+        <div t-att-class="widget.widget_class">
+            <table class="oe_list_content">
+                <thead>
+                    <tr class="oe_list_header_columns">
+                        <th />
+                        <th t-foreach="widget.get_x_axis_values()" t-as="x" t-att-data-x="x">
+                            <t t-esc="widget.get_x_axis_label(x)" />
+                        </th>
+                        <th t-if="widget.show_row_totals">Total</th>
+                    </tr>
+                </thead>
+                <tbody>
+                    <tr t-foreach="widget.get_y_axis_values()" t-as="y" t-att-data-y="y">
+                        <th><t t-esc="widget.get_y_axis_label(y)" /></th>
+                        <td t-foreach="widget.get_x_axis_values()" t-as="x" t-att-class="'oe_list_field_cell' + (widget.is_numeric ? ' oe_number' : '')">
+                            <span t-att-class="widget.get_xy_value_class()">
+                                <input style="display: none" t-att-data-x="x" t-att-data-y="y" t-att-data-id="widget.get_xy_id(x, y)" t-att-value="widget.format_xy_value(widget.get_xy_value(x, y))" />
```

Comment:
TODO to self: it doesn't make sense to fill in x and y on the input. Better put data-x in the td, and leave data-y in tr, then we can select everything by using `tr[data-y="y"] td[data-x="x"]`